exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetMetadataCache.java

```diff
@@ -445,12 +448,148 @@ public void testMetadataCacheAbsolutePaths() throws Exception {
       test("drop table if exists %s", RELATIVE_PATHS_METADATA);
     }
   }
+  @Test // DRILL-4139
+  public void testBooleanPartitionPruning() throws Exception {
```

Comment:
In addition to "=" comparison, can you please try with "is null" / "is not null" predicate, since one of this PR's objectives is to fix partition pruning dealing null values.  I saw another incorrect query result, although I'm not sure if it's a bug in the base code, or a bug in this PR.

```
create table dfs.tmp.`t6/a` as select col_notexist as mykey from cp.`tpch/region.parquet`;
create table dfs.tmp.`t6/b` as select 100 as mykey from cp.`tpch/region.parquet`;
select mykey from dfs.tmp.t6;
+--------+
| mykey  |
+--------+
| null   |
| null   |
| null   |
| null   |
| null   |
| 100    |
| 100    |
| 100    |
| 100    |
| 100    |
+--------+

select mykey from dfs.tmp.t6 where mykey is null;
+--------+
| mykey  |
+--------+
+--------+

``` 

Notice that the last query returns 0 row, where it should return 5 rows.